### PR TITLE
Check for VPN Interfaces

### PR DESCRIPTION
**Pull request by QDaniel**
Wireguard and TINC Interfaces was defunct

### CODE REVIEW

**Comment by AdSchellevis** [comment]
The OpenVPN hook is just a leftover from the legacy system, which we rather don't extend unless there's a generic change which functions for all same type interfaces. 

Do you experience issues when you add a gateway manually? As far as I understood so far, you should be able to add manual gateways for these types.

**Comment by QDaniel** [comment]
With the commit 93bbe1ee168bbe0297746c2a2f68a5a4a4cd0972

The dynamic wireguard and tinc Gateways are broken.

**Comment by AdSchellevis** [comment]
I understand that both commits lead to the same result, but I'm not convinced that we need to re-add something similar now (there might be side affects).

What does your setup look like? Is the gateway address unknown?

**Comment by QDaniel** [comment]
I don't have find a model dir the Gateways to add a setting to Override the defunct check

**Comment by AdSchellevis** [comment]
there isn't. Maybe if we go one step back, we could determine what's needed here. Hence the question about your setup.

**Comment by QDaniel** [comment]
Yes, tinc and WG  are dynamic Interfaces without a Gateway ip

**Comment by AdSchellevis** [comment]
In most situations you know one end of the tunnel, which is likely what most people use. In some cases the address to use is collected dynamically. Sending traffic to an interface without a destination usually isn't needed.

You could always ask @mimugmail what he configures for Wireguard.

**Comment by mimugmail** [comment]
https://forum.opnsense.org/index.php?topic=15105.msg71616#msg71616

**Comment by AdSchellevis** [comment]
I don't mind if we do need to reevaluate the choice only support valid addresses in gateways, but I really need more context about when and why we would like to allow this. An extra checkbox might also be an option in this case, or should we do so for all tunnel types.

**Comment by QDaniel** [comment]
> https://forum.opnsense.org/index.php?topic=15105.msg71616#msg71616

I will check it out and test it. Then we need this only for TINC.
Yes a checkbox for no defunct check where the best.


Yes, we have  4 WG VPNs and 1 TINC .
All wg interfaces are allowed 0.0.0.0/0 so we must set routing manual.
With Static Routes and dynamic Gateways.
TINC is in bridge mode , the routes must be only set to the interface.

**Comment by AdSchellevis** [comment]
@QDaniel @mimugmail maybe we better add an option on the interface, like  https://github.com/opnsense/core/commit/dba70c0ead9b55e4be8ac37dad5c2afbfe3209e8, so we can support routes without intermediate gateways, but don't try to hardcode every tunnel type in the world.

File: src/opnsense/mvc/app/library/OPNsense/Routing/Gateways.php [hunk 1]
```diff
@@ -267,9 +281,11 @@ public function getGateways()
...
                 foreach ($intfgws as $gw_arr) {
                     if (!empty($gw_arr)) {
-                        $gw_arr['disabled'] = true;
-                        $gw_arr['defunct'] = true;
-                        unset($gw_arr['gateway']);
+                        if(!self::IsVPNInterface($gw_arr)){
+                            $gw_arr['disabled'] = true;
+                            $gw_arr['defunct'] = true;
+                            unset($gw_arr['gateway']);
+                        }
```
Comment (drivera73) [inline src/opnsense/mvc/app/library/OPNsense/Routing/Gateways.php]: In some use cases you may want a VPN interface to be your default gateway, so the IsVPNInterface() check here should probably be removed to maintain functional consistency.

Comment (AdSchellevis) [reply src/opnsense/mvc/app/library/OPNsense/Routing/Gateways.php]: @drivera73 this PR won't be included, we're currently opting for this  https://github.com/opnsense/core/commit/dba70c0ead9b55e4be8ac37dad5c2afbfe3209e8

**Comment by QDaniel** [comment]
>  [dba70c0](https://github.com/opnsense/core/commit/dba70c0ead9b55e4be8ac37dad5c2afbfe3209e8)

This is good!

**Pull request closed without merging**